qt/component-box.h

```diff
@@ -40,6 +40,39 @@ class APPSTREAMQT_EXPORT ComponentBox
     Q_GADGET
 
 public:
+    class iterator {
+    public:
+        bool operator==(iterator other) const { return index == other.index && data == other.data; }
+        bool operator!=(iterator other) const { return index != other.index || data != other.data; }
+        iterator& operator++() { index++; return *this; }
+
+        Component operator*() const {
+            return *data->indexSafe(index);
```

Comment:
This will return `NULL` / `std::nullopt` if `index` is out of bounds. What happens then?
If `index` can never become out of bounds, we can potentially give the iterator access to an unsafe accessor method in future to save the "index within bounds" check that `indexSafe` does.

Reply:
I think an unsafe call is what we want here, I figured it can be done once everything is in place.

Reply:
The unsafe query should probably be a private/protected member of the component box then, so it's not called by accident :-)

Reply:
Moved it to the cpp file, no API leakage.